core/src/main/java/cucumber/runtime/Stats.java

```diff
@@ -94,7 +94,7 @@ public void setEventPublisher(EventPublisher publisher) {
 
     public byte exitStatus(boolean isStrict) {
         byte result = 0x0;
-        if (!failedScenarios.isEmpty() || (isStrict && (!pendingScenarios.isEmpty() || !undefinedScenarios.isEmpty()))) {
+        if (!failedScenarios.isEmpty() || (isStrict && (!pendingScenarios.isEmpty() || !undefinedScenarios.isEmpty() || !ambiguousScenarios.isEmpty()))) {
```

Comment:
Looking at [printNonZeroResultScenarios](https://github.com/cucumber/cucumber-jvm/pull/1342/files#diff-9c2af73a8f4e34a02a035f2709e487d5R157) I think `ambiguousScenarios` should be treated the same as `failedScenarios` and always return a non zero exit status.

Reply:
@brasmusson This section feels like it's duplicating some functionality we have elsewhere already. I think we can improve this by using [Result.SEVERITY](https://github.com/cucumber/cucumber-jvm/pull/1323/files#diff-9d1a886ccf16aa0bf49cc278dd2604d0R9) from #1323 to find the worst the results and use [Result.isOk(strict)](https://github.com/cucumber/cucumber-jvm/pull/1323/files#diff-9d1a886ccf16aa0bf49cc278dd2604d0R80) to determine the exit code status.

Reply:
Yes, it sure feels as duplication. Somehow, both here and in `printNonZeroResultScenarios`, `Result.isOk` should be used (instead of duplicating its logic).